packages/lib/src/vis/heatmap/HeatmapMaterial.tsx

```diff
@@ -91,6 +99,12 @@ function HeatmapMaterial(props: Props) {
       withAlpha: alphaValues ? 1 : 0,
       alphaMin: alphaDomain[0],
       oneOverAlphaRange: 1 / (alphaDomain[1] - alphaDomain[0]),
+      nanColor: new Vector4(
+        badColorAsRgb.r / 255,
+        badColorAsRgb.g / 255,
+        badColorAsRgb.b / 255,
+        badColorAsRgb.opacity ?? 1
```

Comment:
not sure it's worth falling back to `1` since `rgb()` seems to take care of initialising the opacity to 1 by default -e.g. for named colours: https://github.com/d3/d3-color/blob/958249d3a17aaff499d2a9fc9a0f7b8b8e8a47c8/src/color.js#L221

and if the user provides an `Rgb` instance directly, the opacity is initialised with `+opacity` in the constructor, which means it always ends up being a number or `NaN` (and if it's `NaN`, it's a programmer error so we shouldn't fallback silently).

Reply:
Excellent remark :+1: